src/sentry/services/hybrid_cloud/organization/__init__.py

```diff
@@ -21,14 +21,21 @@ def get_organization_by_id(
         """
         pass
 
+    # TODO: This should return ApiOrganizationSummary objects, since we cannot realistically span out requests and
+    #  capture full org objects / teams / permissions.  But we can gather basic summary data from the control silo.
     @abstractmethod
     def get_organizations(
-        self, user_id: Optional[int], scope: Optional[str], only_visible: bool
+        self,
+        user_id: Optional[int],
+        scope: Optional[str],
+        only_visible: bool,
+        organization_ids: Optional[List[int]] = None,
     ) -> List["ApiOrganization"]:
         """
-        When user_id is set, returns all organization and membership data associated with that user id given
-        a scope and visibility requirement.  When user_id is None, provides all organizations across the entire
-        system.
+        When user_id is set, returns all organizations associated with that user id given
+        a scope and visibility requirement.  When user_id is not set, but organization_ids is, provides the
+        set of organizations matching those ids, ignore scope and user_id.
```

Comment:
Should there be assertions to ensure that when `organization_ids` is used that `user_id` is None?